libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ConfirmInput.cs

```diff
@@ -189,5 +190,29 @@ private string DetermineCulture(DialogContext dc)
 
             return culture;
         }
+
+        private async Task<ChoiceSet> GetConfirmChoicesAsync(DialogContext dc, (Choice, Choice, ChoiceFactoryOptions) defaults)
+        {
+            ChoiceSet confirmChoices = null;
+            if (ConfirmChoices != null)
+            {
+                if (ConfirmChoices.ExpressionText != null && ConfirmChoices.ExpressionText.TrimStart().StartsWith("${", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
As much as i like a ready to go solution, but i'm a little bit concern of the implication here. 

This is essentially using ${} to differentiae LG vs Expression. So, the message to user will be, when defining a property there are two syntax to referring to LG templates(functions) and built-in functions, 
* ${} for LG functions, like ${myChoices()}
* = for expression functions, like =array("a," "b")

I feel this would probably be confusing to user, and i don't think the direction we are going is to make a difference between LG and Expression.

Reply:
I agree with Dong that this is likely to be confusing.  Alternatively if you used ">! # exports = templates..." then you could access LG templates in expressions.

---
In reply to: [592885731](https://github.com/microsoft/botbuilder-dotnet/pull/5309#discussion_r592885731) [](ancestors = 592885731)